### ISSUE TITLE
Bug when Compiling rust-mozjs

### ISSUE THREAD

**Issue by Redfire75369** [mozilla/sccache#1250]
I seem to be getting a weird error when running `cargo build --release -vv` on [rust-mozjs](https://github.com/servo/rust-mozjs) and I'm not sure why.
```
sccache: encountered fatal error
sccache: error: Not an archive file (invalid global header)
sccache: caused by: Not an archive file (invalid global header)
```
Here's a gist with the full output: https://gist.github.com/Redfire75369/679cbd87bae84cd8eb509647ab1ab1f3
Here's a gist when running without `-vv`: https://gist.github.com/Redfire75369/840a10759351b1bb8970c820da6f7a5f

The build works fine without `sccache` but this happens when `sccache` is added as `build.rustc-wrapper` and `set CCACHE=sccache`.

**UPDATE**: I tested with only `CCACHE=sccache` and removing `build.rustc-wrapper="sccache" and the build succeeds.

**Comment by drahnr** [comment]
Could you enable debug logs on the client/your laptop before starting sccache the first time/kill the spawned process on the client. This doesn't make sense to me yet, any decompression errors should be handled as cache misses on the client https://github.com/mozilla/sccache/blob/7897308c85f435f917cb60b334cecb84f621cfac/src/compiler/compiler.rs#L277 so I'd like to find the path to this error message.

**Comment by Redfire75369** [comment]
Enabling debug logs just requires `SCCACHE_LOG=debug`, right?

**Comment by drahnr** [comment]
https://github.com/mozilla/sccache#debugging - I'd recommend to just run the server in a separate pane using `SCCACHE_START_SERVER=1 SCCACHE_NO_DAEMON=1` (only in the daemon terminal) and collect the logs of both (both should have the `SCCACHE_LOG=debug` set) the local daemon process and the rustc wrapping process. Once both are there, we should have better picture.

**Comment by Redfire75369** [comment]
Weirdly I can't replicate this anymore, and I'm now very confused.
The errors are also weird and don't show any sign of working one way or the other.

**Comment by drahnr** [comment]
Hmm, I can't really do much without additional information. If you happen to encounter it again, please feel free to re-open or create another issue. I suspect it's during toolchain upload but I can't verify at this point.

**Comment by sagudev** [comment]
I have been able to replicate it here: https://github.com/sagudev/mozjs/actions/runs/4698708264.
Sccache log is available as artifact.

Workflow file: https://github.com/sagudev/mozjs/actions/runs/4698708264/workflow#L43

**Comment by drahnr** [comment]
```
D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/infback.c(632,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateBackEnd(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(32,24): note: expanded from macro 'inflateBackEnd'
  #define inflateBackEnd MOZ_Z_inflateBackEnd
                         ^
  5 warnings generated.
  [2023-04-14T10:18:28Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:28Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:28Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:28Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:28Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:28Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inffast.c(50,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL inflate_fast(strm, start)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(77,22): note: expanded from macro 'inflate_fast'
  #define inflate_fast MOZ_Z_inflate_fast
                       ^
  1 warning generated.
  [2023-04-14T10:18:28Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:28Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:28Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(105,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local int inflateStateCheck(strm)
            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(156,27): note: expanded from macro 'inflateStateCheck'
  #define inflateStateCheck MOZ_Z_inflateStateCheck
                            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(105,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(156,27): note: expanded from macro 'inflateStateCheck'
  #define inflateStateCheck MOZ_Z_inflateStateCheck
                            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(119,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateResetKeep(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(117,26): note: expanded from macro 'inflateResetKeep'
  #define inflateResetKeep MOZ_Z_inflateResetKeep
                           ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(145,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateReset(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(30,22): note: expanded from macro 'inflateReset'
  #define inflateReset MOZ_Z_inflateReset
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(158,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateReset2(strm, windowBits)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(111,23): note: expanded from macro 'inflateReset2'
  #define inflateReset2 MOZ_Z_inflateReset2
                        ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(196,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateInit2_(strm, windowBits, version, stream_size)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(61,23): note: expanded from macro 'inflateInit2_'
  #define inflateInit2_ MOZ_Z_inflateInit2_
                        ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(240,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateInit_(strm, version, stream_size)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(60,22): note: expanded from macro 'inflateInit_'
  #define inflateInit_ MOZ_Z_inflateInit_
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(248,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflatePrime(strm, bits, value)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(93,22): note: expanded from macro 'inflatePrime'
  #define inflatePrime MOZ_Z_inflatePrime
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(279,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void fixedtables(state)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(279,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(397,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local int updatewindow(strm, end, copy)
            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(155,22): note: expanded from macro 'updatewindow'
  #define updatewindow MOZ_Z_updatewindow
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(397,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(155,22): note: expanded from macro 'updatewindow'
  #define updatewindow MOZ_Z_updatewindow
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(623,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflate(strm, flush)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(19,17): note: expanded from macro 'inflate'
  #define inflate MOZ_Z_inflate
                  ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1301,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateEnd(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(20,20): note: expanded from macro 'inflateEnd'
  #define inflateEnd MOZ_Z_inflateEnd
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1315,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateGetDictionary(strm, dictionary, dictLength)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(126,30): note: expanded from macro 'inflateGetDictionary'
  #define inflateGetDictionary MOZ_Z_inflateGetDictionary
                               ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1338,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateSetDictionary(strm, dictionary, dictLength)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(27,30): note: expanded from macro 'inflateSetDictionary'
  #define inflateSetDictionary MOZ_Z_inflateSetDictionary
                               ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1373,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateGetHeader(strm, head)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(94,26): note: expanded from macro 'inflateGetHeader'
  #define inflateGetHeader MOZ_Z_inflateGetHeader
                           ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1401,16): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local unsigned syncsearch(have, buf, len)
                 ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(154,20): note: expanded from macro 'syncsearch'
  #define syncsearch MOZ_Z_syncsearch
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1401,16): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(154,20): note: expanded from macro 'syncsearch'
  #define syncsearch MOZ_Z_syncsearch
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1424,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateSync(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(28,21): note: expanded from macro 'inflateSync'
  #define inflateSync MOZ_Z_inflateSync
                      ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1482,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateSyncPoint(strm)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(63,26): note: expanded from macro 'inflateSyncPoint'
  #define inflateSyncPoint MOZ_Z_inflateSyncPoint
                           ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1492,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateCopy(dest, source)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(29,21): note: expanded from macro 'inflateCopy'
  #define inflateCopy MOZ_Z_inflateCopy
                      ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1539,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateUndermine(strm, subvert)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(112,26): note: expanded from macro 'inflateUndermine'
  #define inflateUndermine MOZ_Z_inflateUndermine
                           ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1557,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT inflateValidate(strm, check)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1572,14): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  long ZEXPORT inflateMark(strm)
               ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(110,21): note: expanded from macro 'inflateMark'
  #define inflateMark MOZ_Z_inflateMark
                      ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inflate.c(1585,23): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  unsigned long ZEXPORT inflateCodesUsed(strm)
                        ^
  26 warnings generated.
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/inftrees.c(32,19): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZLIB_INTERNAL inflate_table(type, lens, codes, table, bits, work)
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(78,23): note: expanded from macro 'inflate_table'
  #define inflate_table MOZ_Z_inflate_table
                        ^
  1 warning generated.
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(379,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL _tr_init(s)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(72,18): note: expanded from macro '_tr_init'
  #define _tr_init MOZ_Z__tr_init
                   ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(407,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void init_block(s)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(162,20): note: expanded from macro 'init_block'
  #define init_block MOZ_Z_init_block
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(407,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(162,20): note: expanded from macro 'init_block'
  #define init_block MOZ_Z_init_block
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(451,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void pqdownheap(s, tree, k)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(163,20): note: expanded from macro 'pqdownheap'
  #define pqdownheap MOZ_Z_pqdownheap
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(451,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(163,20): note: expanded from macro 'pqdownheap'
  #define pqdownheap MOZ_Z_pqdownheap
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(486,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void gen_bitlen(s, desc)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(486,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(572,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void gen_codes (tree, max_code, bl_count)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(572,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(615,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void build_tree(s, desc)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(160,20): note: expanded from macro 'build_tree'
  #define build_tree MOZ_Z_build_tree
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(615,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(160,20): note: expanded from macro 'build_tree'
  #define build_tree MOZ_Z_build_tree
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(703,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void scan_tree (s, tree, max_code)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(164,19): note: expanded from macro 'scan_tree'
  #define scan_tree MOZ_Z_scan_tree
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(703,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(164,19): note: expanded from macro 'scan_tree'
  #define scan_tree MOZ_Z_scan_tree
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(748,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void send_tree (s, tree, max_code)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(165,19): note: expanded from macro 'send_tree'
  #define send_tree MOZ_Z_send_tree
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(748,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(165,19): note: expanded from macro 'send_tree'
  #define send_tree MOZ_Z_send_tree
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(799,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local int build_bl_tree(s)
            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(799,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(834,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void send_all_trees(s, lcodes, dcodes, blcodes)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(834,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(863,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL _tr_stored_block(s, buf, stored_len, last)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(73,26): note: expanded from macro '_tr_stored_block'
  #define _tr_stored_block MOZ_Z__tr_stored_block
                           ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(887,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL _tr_flush_bits(s)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(124,24): note: expanded from macro '_tr_flush_bits'
  #define _tr_flush_bits MOZ_Z__tr_flush_bits
                         ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(897,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL _tr_align(s)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(70,19): note: expanded from macro '_tr_align'
  #define _tr_align MOZ_Z__tr_align
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(912,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL _tr_flush_block(s, buf, stored_len, last)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(71,25): note: expanded from macro '_tr_flush_block'
  #define _tr_flush_block MOZ_Z__tr_flush_block
                          ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1015,19): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZLIB_INTERNAL _tr_tally (s, dist, lc)
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(74,19): note: expanded from macro '_tr_tally'
  #define _tr_tally MOZ_Z__tr_tally
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1043,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void compress_block(s, ltree, dtree)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(161,24): note: expanded from macro 'compress_block'
  #define compress_block MOZ_Z_compress_block
                         ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1043,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(161,24): note: expanded from macro 'compress_block'
  #define compress_block MOZ_Z_compress_block
                         ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1103,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local int detect_data_type(s)
            ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1103,11): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1137,16): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local unsigned bi_reverse(code, len)
                 ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1137,16): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1152,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void bi_flush(s)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(157,18): note: expanded from macro 'bi_flush'
  #define bi_flush MOZ_Z_bi_flush
                   ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1152,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(157,18): note: expanded from macro 'bi_flush'
  #define bi_flush MOZ_Z_bi_flush
                   ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1169,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  local void bi_windup(s)
             ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(158,19): note: expanded from macro 'bi_windup'
  #define bi_windup MOZ_Z_bi_windup
                    ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/trees.c(1169,12): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(158,19): note: expanded from macro 'bi_windup'
  #define bi_windup MOZ_Z_bi_windup
                    ^
  34 warnings generated.
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/uncompr.c(27,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT uncompress2 (dest, destLen, source, sourceLen)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(167,21): note: expanded from macro 'uncompress2'
  #define uncompress2 MOZ_Z_uncompress2
                      ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/uncompr.c(86,13): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  int ZEXPORT uncompress (dest, destLen, source, sourceLen)
              ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(37,20): note: expanded from macro 'uncompress'
  #define uncompress MOZ_Z_uncompress
                     ^
  2 warnings generated.
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/zutil.c(133,22): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  const char * ZEXPORT zError(err)
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(65,16): note: expanded from macro 'zError'
  #define zError MOZ_Z_zError
                 ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/zutil.c(305,22): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  voidpf ZLIB_INTERNAL zcalloc (opaque, items, size)
                       ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(80,17): note: expanded from macro 'zcalloc'
  #define zcalloc MOZ_Z_zcalloc
                  ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/zutil.c(315,20): warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
  void ZLIB_INTERNAL zcfree (opaque, ptr)
                     ^
  D:/a/mozjs/mozjs/mozjs/mozjs/modules/zlib/src/mozzconf.h(81,16): note: expanded from macro 'zcfree'
  #define zcfree MOZ_Z_zcfree
                 ^
  3 warnings generated.
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:29Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:29Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:29Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:30Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:30Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:30Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:31Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:31Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:31Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:32Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:32Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:32Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:32Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:32Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:32Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:32Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:32Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:32Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:33Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:33Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:33Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:33Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:33Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:33Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:33Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:33Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:33Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:33Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:33Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:33Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:34Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:34Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:34Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:34Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:34Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:34Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:34Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:34Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:34Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:34Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:34Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:34Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:34Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:34Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:34Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:35Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:35Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:35Z DEBUG sccache::commands] Server sent CompileStarted
  [2023-04-14T10:18:35Z DEBUG sccache::config] Attempting to read config file at "C:\\Users\\runneradmin\\AppData\\Roaming\\Mozilla\\sccache\\config\\config"
  [2023-04-14T10:18:35Z DEBUG sccache::config] Couldn't open config file: failed to open file `C:\Users\runneradmin\AppData\Roaming\Mozilla\sccache\config\config`
  [2023-04-14T10:18:35Z DEBUG sccache::commands] Server sent CompileStarted


  error occurred: Command "sccache" "clang-cl" "-nologo" "-MD" "-Z7" "-Brepro" "-m64" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\dist/include" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\js/src" "-W4" "-FI" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build/js/src/js-confdefs.h" "-FoD:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\glue\\src/jsglue.o" "-c" "--" "src/jsglue.cpp" with args "clang-cl" did not execute successfully (status code exit code: 1).


Error: Process completed with exit code 101.
```

is the relevant part

**Comment by drahnr** [comment]
@Xuanwo  @sylvestre I have no windows machine or capacity to dig into this

**Comment by sylvestre** [comment]
same. maybe @glandium knows

**Comment by Redfire75369** [comment]
For me at least, this still works locally. With both `RUSTC_WRAPPER=sccache` and `CCACHE=sccache`, it compiles fine. It builds the whole thing with no errors.

**Comment by Redfire75369** [comment]
This is very weird, so it's not working now, although it was previously (I did `cargo clean`, before you ask).
My current problem is this, which seems to be because `RUST_BINDGEN` isn't being defined? (See <https://searchfox.org/mozilla-central/source/js/src/js-config.h.in#25-29>)
```
[mozjs_sys 0.68.2] cargo:warning=In file included from src/jsglue.cpp:7:
[mozjs_sys 0.68.2] cargo:warning=In file included from src/jsglue.hpp:10:
[mozjs_sys 0.68.2] cargo:warning=In file included from C:\Users\Redfire\software\github\mozjs\target\debug\build\mozjs_sys-33e19f40c6cf1c50\out\build\dist/include\js/ArrayBuffer.h:14:
[mozjs_sys 0.68.2] cargo:warning=In file included from C:\Users\Redfire\software\github\mozjs\target\debug\build\mozjs_sys-33e19f40c6cf1c50\out\build\dist/include\jstypes.h:32:
[mozjs_sys 0.68.2] cargo:warning=C:\Users\Redfire\software\github\mozjs\target\debug\build\mozjs_sys-33e19f40c6cf1c50\out\build\dist/include/js-config.h(27,4): error: "SpiderMonkey was configured with --enable-debug, so DEBUG must be defined when including this header"
[mozjs_sys 0.68.2] cargo:warning=#  error "SpiderMonkey was configured with --enable-debug, so DEBUG must be defined when including this header"
[mozjs_sys 0.68.2] cargo:warning=   ^
[mozjs_sys 0.68.2] cargo:warning=1 error generated.
```

Seems that the `-DRUST_BINDGEN` is getting eaten up somehow, when `RUSTC_WRAPPER=sccache`.
<https://github.com/servo/mozjs/blob/master/mozjs/build.rs#L98>

**Comment by Redfire75369** [comment]
Both the flag checks passed fine, but I see `-DSTATIC_JS_API` in the `-vv` output but not `-DRUST_BINDGEN`.
```
handle_client: compile
check_compiler: Supported compiler
parse_arguments: Ok: ["-nologo", "-MD", "-Brepro", "-m64", "-W4", "-DRUST_BINDGEN", "-FoC:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check", "-c", "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp"]
[flag_check.obj]: get_cached_or_compile: ["-nologo", "-MD", "-Brepro", "-m64", "-W4", "-DRUST_BINDGEN", "-FoC:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check", "-c", "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp"]
preprocess: Some("C:\\Program Files\\LLVM\\bin\\clang-cl.exe" "-EP" "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp" "-nologo" "-DRUST_BINDGEN" "-nologo" "-MD" "-Brepro" "-m64" "-W4")
[flag_check.obj]: generate_hash_key took 0.646 s
[flag_check.obj]: Cache hit in 0.000 s
handle_client: compile
check_compiler: Supported compiler
parse_arguments: Ok: ["-nologo", "-MD", "-Brepro", "-m64", "-W4", "-DSTATIC_JS_API", "-FoC:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check", "-c", "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp"]
[flag_check.obj]: get_cached_or_compile: ["-nologo", "-MD", "-Brepro", "-m64", "-W4", "-DSTATIC_JS_API", "-FoC:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check", "-c", "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp"]
preprocess: Some("C:\\Program Files\\LLVM\\bin\\clang-cl.exe" "-EP" "C:\\Users\\Redfire\\software\\github\\mozjs\\target\\debug\\build\\mozjs_sys-33e19f40c6cf1c50\\out\\flag_check.cpp" "-nologo" "-DSTATIC_JS_API" "-nologo" "-MD" "-Brepro" "-m64" "-W4")
[flag_check.obj]: generate_hash_key took 0.379 s
[flag_check.obj]: Cache hit in 0.000 s
```

**Comment by Redfire75369** [comment]
As for the issue with the initial run (no `--features debugmozjs` which I was using), `-std=c++17` is getting eaten up instead so `std::in_place_t` among other errors, which prevents the compilation.
I don't understand why flags are being eaten up, but that seems to be the root of the issue.

Notice there is no `-std=c++17` here
```
running: "sccache" "clang-cl" "-nologo" "-MD" "-Z7" "-Brepro" "-m64" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\dist/include" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\js/src" "-W4" "-FI" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build/js/src/js-confdefs.h" "-FoD:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\glue\\src/jsglue.o" "-c" "--" "src/jsglue.cpp"
```

**Comment by drahnr** [comment]
Some dependencies on env vars `cargo:rerun-if-env-change` are behind conditionals. So if you had the `.._RERUN_..` env set, they are not dependencies, and hence are not included in the cli parameters and hence are not detected in `sccache`.
Fix: Always make them dependencies

**Comment by drahnr** [comment]
@Redfire75369 @sagudev gentle ping?

**Comment by sagudev** [comment]
I tried moving [`cargo:rerun-if-env-change` out of if statements](https://github.com/sagudev/mozjs/commit/468ceeb0f2d61530a7f426f92a766c20443ccda8), but [builds are still failing](https://github.com/sagudev/mozjs/actions/runs/4732721507).

**Comment by drahnr** [comment]
> I tried moving [`cargo:rerun-if-env-change` out of if statements](https://github.com/sagudev/mozjs/commit/468ceeb0f2d61530a7f426f92a766c20443ccda8), but [builds are still failing](https://github.com/sagudev/mozjs/actions/runs/4732721507).

Could you copy the relevant sections / calls in here including relevant env items?

**Comment by sagudev** [comment]
There are now these env vars printing unconditionally:
```
cargo:rerun-if-env-changed=AR
cargo:rerun-if-env-changed=AS
cargo:rerun-if-env-changed=CC
cargo:rerun-if-env-changed=CFLAGS
cargo:rerun-if-env-changed=CLANGFLAGS
cargo:rerun-if-env-changed=CPP
cargo:rerun-if-env-changed=CPPFLAGS
cargo:rerun-if-env-changed=CXX
cargo:rerun-if-env-changed=CXXFLAGS
cargo:rerun-if-env-changed=MAKE
cargo:rerun-if-env-changed=MOZ_TOOLS
cargo:rerun-if-env-changed=MOZJS_FORCE_RERUN
cargo:rerun-if-env-changed=MOZTOOLS_PATH
cargo:rerun-if-env-changed=PYTHON
cargo:rerun-if-env-changed=STLPORT_LIBS
```
Failing on:
```
error occurred: Command "sccache" "clang-cl" "-nologo" "-MD" "-Z7" "-Brepro" "-m64" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\dist/include" "-I" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\js/src" "-W4" "-FI" "D:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build/js/src/js-confdefs.h" "-FoD:\\a\\mozjs\\mozjs\\target\\debug\\build\\mozjs_sys-e87172a29422a6d2\\out\\build\\glue\\src/jsglue.o" "-c" "--" "src/jsglue.cpp" with args "clang-cl" did not execute successfully (status code exit code: 1).
```

Do note that same env vars arre conditioned also on linux and mac builds, but they run successfully with sccache as rustc wrapper.

**Comment by alex** [comment]
FWIW, attempting to use sccache in a mixed python/rust project, we're seeing this on all of our macOS builders: https://github.com/pyca/cryptography/actions/runs/4931490682. All the other builders look ok though.

**Comment by alex** [comment]
Ok, I've got a minimal reproducer:

`Cargo.toml`:
```
[package]
name = "openssl-universal"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
openssl-sys = "0.9.87"
```

`src/main.rs`:
```
fn main() {
    println!("Hello, world!");
}
```

Build steps:

```
~/p/h/openssl-universal ❯❯❯ RUSTC_WRAPPER=sccache OPENSSL_DIR=(pwd)/ossl/ cargo check
    Checking openssl-sys v0.9.87
sccache: encountered fatal error
sccache: error: Not an archive file (invalid global header)
sccache: caused by: Not an archive file (invalid global header)
error: could not compile `openssl-sys`

Caused by:
  process didn't exit successfully: `sccache rustc --crate-name openssl_sys --edition=2018 /Users/alex_gaynor/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-sys-0.9.87/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=135 --crate-type lib --emit=dep-info,metadata -C embed-bitcode=no -C split-debuginfo=unpacked -C debuginfo=2 -C metadata=12560415109ccd92 -C extra-filename=-12560415109ccd92 --out-dir /Users/alex_gaynor/projects/hack/openssl-universal/target/debug/deps -L dependency=/Users/alex_gaynor/projects/hack/openssl-universal/target/debug/deps --extern libc=/Users/alex_gaynor/projects/hack/openssl-universal/target/debug/deps/liblibc-f23871325ead10a5.rmeta --cap-lints allow -L native=/Users/alex_gaynor/projects/hack/openssl-universal/ossl/lib -l static=ssl -l static=crypto --cfg openssl --cfg 'osslconf="OPENSSL_NO_COMP"' --cfg 'osslconf="OPENSSL_NO_SSL3_METHOD"' --cfg ossl300 --cfg ossl101 --cfg ossl102 --cfg ossl102f --cfg ossl102h --cfg ossl110 --cfg ossl110f --cfg ossl110g --cfg ossl110h --cfg ossl111 --cfg ossl111b --cfg ossl111c` (exit status: 254)
```

`ossl/` dir contains an unzipped copy of the universal2 OpenSSL from the latest build on https://github.com/pyca/infra/actions/workflows/build-macos-openssl.yml

**Comment by Xuanwo** [comment]
I'm trying to fix this issue by https://github.com/mozilla/sccache/pull/1773, please give it a try.
